TestClients/js/email/send-email.js

```diff
@@ -0,0 +1,58 @@
+const { EmailClient } = require("@azure/communication-email");
+require("dotenv").config();
+
+// This code demonstrates how to fetch your connection string
+// from an environment variable.
+const connectionString = process.env['COMMUNICATION_SERVICES_CONNECTION_STRING'];
+const emailClient = new EmailClient(connectionString);
+
+async function main() {
+    const POLLER_WAIT_TIME = 10
+    try {
+      const message = {
+        senderAddress: "donotreply@azurecomm.net",
+        content: {
+          subject: "Welcome to Azure Communication Services Email",
+          plainText: "This email message is sent from Azure Communication Services Email using the JavaScript SDK.",
+        },
+        recipients: {
+          to: [
+            {
+                address: "ageon-d-conqueror@sevenkingdom.com",
+                displayName: "Aegon Targaryen",
+            },
+          ],
+        },
+      };
+  
+      const poller = await emailClient.beginSend(message);
+  
+      if (!poller.getOperationState().isStarted) {
+        throw "Poller was not started."
+      }
+  
+      let timeElapsed = 0;
+      while(!poller.isDone()) {

```

Comment:
Any reason why you're not using `await poller.pollUntilDone()` instead of manual polling?

Reply:
I followed the instructions in quickstart => https://learn.microsoft.com/en-us/azure/communication-services/quickstarts/email/send-email?tabs=windows%2Cconnection-string&pivots=programming-language-javascript

which had manual polling and didn't realize there is automate polling. I started testing with automated polling and see it as **_unreliable compared to manual polling_**. Manual polling is much controlled, and I can see each time it calls GetOperation API to get status of email via logging whereas I see no logs on how many times it polled and when I set debug, I see exactly two times it calls GetOperation API and if status is successful then sometimes it exits or it just print operationId and hangs. If it doesn't able get success status in two calls/polling then it just hangs, and I have to kill the test. Here is my test log:

![image](https://github.com/DominikMe/acs-emulator/assets/30486500/5338fb0f-4bdc-43ad-a43f-16b14341e4cd)

Reply:
😮 

If `await poller.pollUntilDone()` frequently hangs, this is suspicious! It is a common Azure SDK functionality, unlikely that it has bugs, more probably there is an issue with the emulator API somehow.

We should also work to fix the sample/docs to make it use automatic poller. The automatic poller is there to make everyone's life easier... If it doesn't make life easier but breaks things instead, we should find the issue and fix it.

Reply:
Only thing I think of is if HTTP connection is not long enough to keep polling. As I said, there is no call after first two call from poller so I don't think there is any issue with API. I can fix this by making sure email status is successful after the second call.

There is also case where it hangs even after successful email status (it prints the log of `Successfully sent the email (operation id: ... )` ) and it doesn't exit the test.

It is strange that manual polling works all the time but automatic polling is having so much issues.